### PR TITLE
feat(lib): add a warning about id fields

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

File: packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts [hunk 1]
```diff
@@ -38,12 +54,14 @@ export class StructEmitter {
         this.code.line(
           `* Docs at Terraform Registry: {@link ${resource.linkToDocs}#${att.terraformName} ${resource.className}#${att.terraformName}}`
         );
+        this.warnAboutIdField(att);
```
Comment (ansgarm) [inline packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts:57]: Is there a way we can figure out which provider sdk version has been used? Probably not, right?

Comment (DanielMSchmidt) [reply packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts:57]: Unfortunately this information is not part of the schema, I think one would need to monkey type it, e.g. if there is a nested type present somewhere it's a newer one. Not sure if it's worth the effort though

Comment (ansgarm) [reply packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts:57]: Ah, yeah, not worth it then 😇

File: packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts [hunk 2]
```diff
@@ -19,6 +20,21 @@ export class StructEmitter {
...
   }
 
+  // Due to https://github.com/hashicorp/terraform-plugin-sdk/commit/2387eb85e32c064b4a62718c9f5c80bf00dc7fb9 all
+  // resources from providers using the old SDK have the id field by default
+  // We have no way to distinguish them through the provider schema, so a word of warning for our users
+  private warnAboutIdField(att: AttributeModel) {
+    if (att.name === "id") {
+      this.code.line(`*`);
+      this.code.line(
+        `* Please be aware that the id field is automatically added to all resources in providers using a SDK version below 2.`
```
Comment (ansgarm) [inline packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts:30]:
```suggestion
        `* Please be aware that the id field is automatically added to all resources in Terraform providers using a Terraform provider SDK version below 2.`
```

Comment (ansgarm) [reply packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts:30]: Make it a bit more clear, so users don't wonder which SDK version exactly (if they are not familiar with what happens under the hood)

**Review by ansgarm — APPROVED** [review]

**Pull request merged**